src/ng/compile.js

```diff
@@ -2245,43 +2337,58 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
               directiveNormalize(nodeName), 'E', maxPriority, ignoreDirective);
 
           // iterate over the attributes
-          for (var attr, name, nName, ngAttrName, value, isNgAttr, nAttrs = node.attributes,
+          for (var attr, name, nName, ngAttrName, value, ngPrefixMatch, nAttrs = node.attributes,
                    j = 0, jj = nAttrs && nAttrs.length; j < jj; j++) {
             var attrStartName = false;
             var attrEndName = false;
 
+            var isNgAttr = false, isNgProp = false, isNgEvent = false;
+            var multiElementMatch;
+
             attr = nAttrs[j];
             name = attr.name;
             value = attr.value;
 
-            // support ngAttr attribute binding
+            // support ng-attr-*, ng-prop-* and ng-on-*
             ngAttrName = directiveNormalize(name);
-            isNgAttr = NG_ATTR_BINDING.test(ngAttrName);
-            if (isNgAttr) {
+            if (ngPrefixMatch = ngAttrName.match(NG_PREFIX_BINDING)) {
+              isNgAttr = ngPrefixMatch[1] === "Attr";
+              isNgProp = ngPrefixMatch[1] === "Prop";
+              isNgEvent = ngPrefixMatch[1] === "On";
+
               name = name.replace(PREFIX_REGEXP, '')
-                .substr(8).replace(/_(.)/g, function(match, letter) {
+                .substr(4 + ngPrefixMatch[1].length).replace(/_(.)/g, function(match, letter) {
                   return letter.toUpperCase();
                 });
-            }
 
-            var multiElementMatch = ngAttrName.match(MULTI_ELEMENT_DIR_RE);
-            if (multiElementMatch && directiveIsMultiElement(multiElementMatch[1])) {
+            // support *-start / *-end multi element directives
+            } else if ((multiElementMatch = ngAttrName.match(MULTI_ELEMENT_DIR_RE)) && directiveIsMultiElement(multiElementMatch[1])) {
               attrStartName = name;
               attrEndName = name.substr(0, name.length - 5) + 'end';
               name = name.substr(0, name.length - 6);
             }
 
             nName = directiveNormalize(name.toLowerCase());
             attrsMap[nName] = name;
```

Comment:
Is this right for ng-on/prop? I don't think we'd want to "map" `ng-on-foo` to the `foo` attribute... Would we want to map it at all? To `ngAttrName` (still containing ng-on/prop prefix) instead of `name`?

Reply:
It is fine imo. We should indeed not reflect the property/event as attribute, but keeping track of the `ngProp/On` prefixed attribute sounds reasonable and consistent (since the attribute is there).

Reply:
I think your comment is true regarding the `attrs[ngAttrName] = value;` a few lines down. But what about the "mapping" on this line?

This line creates the `ngPropFoo => foo` mapping for https://docs.angularjs.org/api/ng/type/$compile.directive.Attributes#$attr ...

Reply:
Oh, you are right. I don't think we want to map `foo` to `ngPropFoo`. 
(Mapping `ng-prop-foo` to `ngPropFoo` sounds right (for consistency), although it would probably have no effect.)